internal/batches/ui/interval_writer.go

```diff
@@ -119,6 +119,14 @@ type prefixedWriter struct {
 func (w *prefixedWriter) Write(p []byte) (int, error) {
```

Comment:
I just checked the source of `p` and I think we will only ever stream 1 line at a time, either with the newline at the end, or not if the stream has ended and there is no newline (for example `echo -n`).
So that makes me thing that we shouldn't even split the `p` in here at all - and would as a side-effect get that for free, while retaining _actual_ empty log lines. How does that feel for you?

Reply:
Oh! Yeah true - that's better. I can just remove the loop then. It makes sense to me.

Reply:
Should just re-validate that - but from reading code it seems like our scanner on the other end of this stream would write in line-chunks :)

Reply:
Yeah, I just tested it, and it works. For both SSBC and local executions - the log output is consistent now.
Pushed a refactor.